wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java

```diff
@@ -414,7 +417,7 @@ public void transform(StructuredRecord input, Emitter<StructuredRecord> emitter)
       }
 
       // If pre-condition is set, then evaluate the precondition
-      if (PRECONDITION_LANGUAGE_JEXL.equalsIgnoreCase(config.getPreconditionLanguage())
+      if (LANGUAGE_JEXL.equalsIgnoreCase(config.getExecutionLanguage())
           && checkPreconditionNotEmpty(false)) {
```

Comment:
This logic is executed on every row which is not efficient. Can we execute this logic once and store the result in a global variable?